daemon/api.go

```diff
@@ -457,13 +464,51 @@ func getStore(c *Command) snapstate.StoreService {
 	return snapstate.Store(st)
 }
 
+func getSections(c *Command, r *http.Request, user *auth.UserState) Response {
+	route := c.d.router.Get(snapCmd.Path)
+	if route == nil {
+		return InternalError("cannot find route for snaps")
+	}
+
+	theStore := getStore(c)
+
+	sections, err := theStore.Sections(user)
+	switch err {
+	case nil:
+		// pass
+	case store.ErrEmptyQuery, store.ErrBadQuery:
+		return BadRequest("%v", err)
+	case store.ErrUnauthenticated:
+		return Unauthorized(err.Error())
+	default:
+		return InternalError("%v", err)
+	}
+
+	results := make([]*json.RawMessage, 0, len(sections))
```

Comment:
I'm missing something here. `sections` is a `[]string`, so why not just return it? As in `return SyncResponse(sections, meta)`?